### PR TITLE
Support alternative default branch names other than master

### PR DESCRIPTION
**Pull request by markjamesbutler**
## What does this change?

It removes the hardcoding of the people branch to master. The Guardian is moving the default branch name used in it's repos from master to main. See [here](https://github.com/guardian/master-to-main) for more detail. 

## How to test

I have written a test against a local repo to verify the change works against a main default repo. 

## How can we measure success?

If the system continues to work for both master and main repos then it will have been successful. 

## Have we considered potential risks?

We cannot fully test it against PROD until we change the people [repo](https://github.com/guardian/people) to default to main. 
We need to be careful of any repos that don't have master set as the default branch.
If the code doesn't work as expected it could break the gu-who system in general. 

## Images
n/a

### CODE REVIEW

**Comment by markjamesbutler** [comment]
@rtyley let's chat about how to deploy this before we merge as it requires synchronisation with updating the people repo default branch.

File: test/lib/PeopleRepoSpec.scala [hunk 1]
```diff
@@ -13,17 +13,30 @@ class PeopleRepoSpec extends PlaySpec {
...
       implicit val demoRemoteRepo = unpackRepo("/demo-people-repos/people.no-users-txt.git.zip")
 
-      val logins = PeopleRepo.getSponsoredUserLogins(Path.createTempDirectory(), demoRemoteRepo.getDirectory.getAbsolutePath)
+      val logins = PeopleRepo.getSponsoredUserLogins(Path.createTempDirectory(), demoRemoteRepo.getDirectory.getAbsolutePath, defaultBranch = "master")
+      logins mustBe empty
+    }
+
+    "extract sponsored user logins after cloning a remote repo with main default branch" in {
+
+      implicit val demoRemoteRepo = unpackRepo("/demo-people-repos/minimal-people-main.git.zip")
```
Comment (rtyley) [inline test/lib/PeopleRepoSpec.scala:30]: The Travis build for this project was long-term broken before, I only just got it working now (!) - when it ran, [the tests failed](https://travis-ci.com/github/guardian/gu-who/builds/222291040#L251) as I think you've introduced a new test resource (`minimal-people-main.git.zip`) but not committed to the PR?

image

What's needed is a tiny zipped test repo, like https://github.com/guardian/gu-who/blob/main/test/resources/demo-people-repos/minimal-people.git.zip - but with the default branch of `main`.

Comment (markjamesbutler) [reply test/lib/PeopleRepoSpec.scala:30]: Thank you @rtyley I've pushed that file now and the build is passing.

**Comment by markjamesbutler** [comment]
@rtyley Are you able to re-approve this now the build is fixed?

**Review by rtyley — APPROVED** [review]
Looks good to me!

**Comment by rtyley** [comment]
> @rtyley let's chat about how to deploy this before we merge as it requires synchronisation with updating the people repo default branch.

When's good to have a chat?

**Comment by BioGeek** [comment]
The [people demo repo](https://github.com/gu-who-demo-org/people) still defaults to `master`, so the link https://github.com/gu-who-demo-org/people/blob/main/users.txt in the [README](https://github.com/guardian/gu-who/blob/main/README.md#enforced-requirements) now gives a 404 error.

**Pull request merged**